### ISSUE TITLE
Add a learn more link on Safe Browsing setting to our support page

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#20514]
We now have a section about Safe Browsing in our privacy policy: https://brave.com/privacy/browser/#safe-browsing

We should link to it from `brave://settings/security` on desktop:
![Screenshot from 2022-01-14 16-33-10](https://user-images.githubusercontent.com/167821/149601481-50937e86-0d76-496c-becf-8603a27a5911.png)

and from *Settings | Brave Shiels & privacy | Safe Browsing* on Android:
![Screenshot_20220114-162951](https://user-images.githubusercontent.com/167821/149601521-31f10faf-0a64-4624-a1d0-1da0b73d942d.png)

I don't know whether this needs input from the design team, but for Desktop, I was thinking of a "Learn more" link like this one:
![Screenshot from 2022-01-14 16-38-53](https://user-images.githubusercontent.com/167821/149601732-f39e744e-bb0f-44b9-b660-7c70b2a4393b.png)

and for Android, perhaps we could simply change what the `(?)` button in the top-right corner links to? Right now, it links to community.brave.com which seems extremely generic and not particularly useful in this context.

**Comment by fmarier** [comment]
Instead of our privacy policy, this should link to https://support.brave.com/hc/en-us/articles/15222663599629-Safe-Browsing-in-Brave.

**Comment by fmarier** [comment]
Safe Browsing intersitials currently link to the [Google support page](https://support.google.com/chrome/answer/99020?visit_id=638185778760175000-3395272883&p=cpn_safe_browsing&rd=1) and so should be updated to point to our page instead.

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.53.45 Chromium: 114.0.5735.26 (Official Build) nightly (x86_64)
-- | --
Revision | 7075cbb66f0542ac3e01ddfde6b813e7d61118a5-refs/branch-heads/5735@{#454}
OS | macOS Version 11.7.6 (Build 20G1231)

### Steps:
1. installed `1.53.45`
2. launched Brave
3. waited for 5 minutes
4. loaded `https://testsafebrowsing.appspot.com/s/malware.html`
5. confirmed interstitial page (and clicked `Details`)
6. confirmed `Learn more` link opened `https://support.brave.com/hc/en-us/articles/15222663599629-Safe-Browsing-in-Brave` in a new tab
7. loaded `https://testsafebrowsing.appspot.com/s/malware.html`
8. clicked on `Back to safety`
9. confirmed it opened the new-tab page
10. opened `brave://settings/security` and clicked on the `(?)` icon on the top right
11. confirmed it opened `https://support.brave.com/hc/en-us/articles/15222663599629-Safe-Browsing-in-Brave` in a new tab


example | example | example | example | example | example
---------|---------|---------|----------|---------|----------
<img width="1426" alt="Screen Shot 2023-05-16 at 8 40 32 PM" src="https://github.com/brave/brave-browser/assets/387249/70c7e367-baf9-405c-aa72-8f7710ab55b5"> | <img width="1426" alt="Screen Shot 2023-05-16 at 8 40 36 PM" src="https://github.com/brave/brave-browser/assets/387249/5b2029f7-6d16-41ba-be8d-5213b8f847e3"> | <img width="1426" alt="Screen Shot 2023-05-16 at 8 43 29 PM" src="https://github.com/brave/brave-browser/assets/387249/4c729281-be9b-4b20-82a7-4e67933692e5"> | <img width="1426" alt="Screen Shot 2023-05-16 at 8 40 53 PM" src="https://github.com/brave/brave-browser/assets/387249/05728782-53d6-41b6-8bbb-2e4cf484a439"> | <img width="1426" alt="Screen Shot 2023-05-16 at 8 40 12 PM" src="https://github.com/brave/brave-browser/assets/387249/d8ffad3b-d82c-4ef1-b920-6da791ed5657"> | <img width="1426" alt="Screen Shot 2023-05-16 at 8 40 23 PM" src="https://github.com/brave/brave-browser/assets/387249/421e4b47-5d34-4b84-89b5-7723f8552a97">

**Comment by vlad-hoff** [comment]
Hi @fmarier I just checked Android with Brave Beta and Nightly and see that it transfers user to https://brave.com/privacy/browser/#safe-browsing

Is that expected for Android?

```
Device/OS: Pixel 7 [panther_beta-user 14 UPB2.230407.019 release-keys]
Brave build: 1.53.83
Chromium: 114.0.5735.110 (Official Build) beta (64-bit)
Revision: 1c828682b85bbc70230a48f5e345489ec447373e-refs/branch-heads/5735_90@{#13}
```

https://github.com/brave/brave-browser/assets/128532965/c90b53be-0c42-482c-9892-2cf0fc6d1e8e

**Comment by fmarier** [comment]
@vlad-hoff Good catch. It looks like this was accidentally reverted in https://github.com/brave/brave-core/commit/b4b7d741b11dbd0d25292a3ad6f1b80d253644c9

cc @samartnik

**Comment by fmarier** [comment]
Android fix in https://github.com/brave/brave-core/pull/18838.

**Comment by kjozwiak** [comment]
> Android fix in [brave/brave-core#18838](https://github.com/brave/brave-core/pull/18838).

Removing `OS/Android` from the above issue as `Android` will be fixed via https://github.com/brave/brave-browser/issues/30933 in `1.54.x`.
